### PR TITLE
Simplify modules separation

### LINKED ISSUE #12767: Simplify modules separation
### Description

After dropping the automatic generation of the `config-api` we want to have a Maven configuration that is easier to handle and maintain to improve speed in doing changes.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by andreaTP**
Resolves #12767

### CODE REVIEW

**Comment by andreaTP** [comment]
Keeping 2 separate commits for clarity (I think it does make sense)

**Review by DGuhr — DISMISSED** [review]
LGTM

File: operator/pom.xml [hunk 1]
```diff
@@ -1,36 +1,277 @@
...
+            <groupId>io.quarkiverse.operatorsdk</groupId>
+            <artifactId>quarkus-operator-sdk-csv-generator-deployment</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <!-- Keycloak -->
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-config-api</artifactId>
+        </dependency>
```
Comment (vmuzikar) [inline operator/pom.xml:115]: AFAICT, we don't need this in the Operator anymore?

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: operator/pom.xml (hunk 1 above)
Comment (andreaTP) [reply operator/pom.xml]: sure, thanks for spotting it, fixed 👍 (now depends on `keycloak-commmon`)

**Review by pedroigor — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]

**Review by DGuhr — APPROVED** [review]

**Pull request merged**